### PR TITLE
chore: Migration to convert formData references

### PR DESCRIPTION
**Pull request by nidhi-nair**
Creating as draft plugins' consumption needs to be updated as well.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 1]
```diff
@@ -4932,4 +4932,376 @@ public void migrateFromRSASha1ToECDSASha2Protocol(MongockTemplate mongockTemplat
...
+         * Migrate unpublished action configuration data.
+         */
+        final Map<String, Object> unpublishedFormData = unpublishedAction.getActionConfiguration().getFormData();
+
+        unpublishedFormData
+                .keySet()
+                .stream()
+                .forEach(k -> {
+                    final Object oldValue = unpublishedFormData.get(k);
+                    unpublishedFormData.put(k, Map.of(
```
[the remark this answers is not in the document]
Comment (nidhi-nair) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:5013]: Yes this should be fine, we're streaming through the key set, and that set is not changing. We're updating the map that uses this key set.

**Comment by nidhi-nair** [comment]
@sumitsum Hey so after I resolved the conflicts from upstream, there have been some changes to the S3 plugin structure. Could you please revisit that merge commit to see if the mapping makes sense?

**Comment by sumitsum** [comment]
> @sumitsum Hey so after I resolved the conflicts from upstream, there have been some changes to the S3 plugin structure. Could you please revisit that merge commit to see if the mapping makes sense?

@nidhi-nair LGTM 👍

**Review by sumitsum — APPROVED** [review]

**Pull request merged**